OpenKeychain/src/main/java/org/sufficientlysecure/keychain/pgp/PgpSecurityConstants.java

```diff
@@ -117,26 +134,53 @@ public static boolean isSecureHashAlgorithm(int id) {
             TeleTrusTNamedCurves.getOID("brainpoolP512r1").getId()
     ));
 
-    public static boolean isSecureKey(CanonicalizedPublicKey key) {
-        switch (key.getAlgorithm()) {
+    static KeySecurityProblem checkForSecurityProblems(CanonicalizedPublicKey key) {
+        // TODO: checks on master key?
```

Comment:
no :)

I suppose what I had in mind was "check primary key, not only signing subkey", but that doesn't really belong here.